File: lib-database-annotation/src/main/java/com/ustadmobile/lib/database/annotation/UmQueryFindByPrimaryKey.java
package com.ustadmobile.lib.database.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.CLASS)
@Target(ElementType.METHOD)

/**
 * Indicates that the given method should auto generate a query method that will select the entity
 * (as per the class type of the first parameter of the method) by it's primary key
 */
public @interface UmQueryFindByPrimaryKey {
}

File: lib-database-annotations/src/commonMain/kotlin/com/ustadmobile/lib/annotation/UmQueryFindByPrimaryKey.kt
package com.ustadmobile.lib.database.annotation

@Retention(AnnotationRetention.BINARY)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation
/**
 * Indicates that the given method should auto generate a query method that will select the entity
 * (as per the class type of the first parameter of the method) by it's primary key
 */
class UmQueryFindByPrimaryKey
